src/librustc_metadata/encoder.rs

```diff
@@ -1170,57 +1169,57 @@ impl EncodeContext<'_, 'tcx> {
                     paren_sugar: trait_def.paren_sugar,
                     has_auto_impl: tcx.trait_is_auto(def_id),
                     is_marker: trait_def.is_marker,
-                    super_predicates: self.lazy(&tcx.super_predicates_of(def_id)),
+                    super_predicates: self.lazy(&*tcx.super_predicates_of(def_id)),
```

Comment:
Why does this reborrow?

Reply:
`&Lrc<T>` not being iterable, I think? I'll have to go back and check which of these you switched to `&[...]` (which would solve the problem, IIRC).

Reply:
`GenericPredicates` is not iterable either =P

Reply:
Oh, sorry, I confused it with an iterable situation. Okay yeah so the problem is I don't want a `&Lrc<T>` because that will result in `Lazy<Lrc<T>>` being returned.
But if the `Lrc` is gone then the `&*` is redundant.

Reply:
The `Lrc` is gone here.